src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<T>(null, value, null);
+            tail = head;
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index must be less than size!");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            head = new Node<T>(null, value, head);
+        } else {
+            Node<T> next = getNode(index);
+            Node<T> node = new Node<>(next.prev, value, next);
+            next.prev.next = node;
+            next.prev = node;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index must be less than size!");
+        }
```

Comment:
you have 3 identical code blocks, extract them to a separate method and add the index param to the exception message